components/org.wso2.carbon.identity.oauth.scope.endpoint/src/main/java/org/wso2/carbon/identity/oauth/scope/endpoint/impl/ScopesApiServiceImpl.java

```diff
@@ -233,4 +246,48 @@ public Response deleteScope(String name) {
         }
         return Response.status(Response.Status.OK).build();
     }
+
+    /**
+     * Build the complete URI to the location of the resource.
+     * Ex: https://localhost:9443/t/<tenant-domain>/identity/oauth2/v1.0/scopes/name/<scope_name>
+     *
+     * @param name of the scope.
+     * @return Fully qualified and complete URI.
+     */
+    private static URI buildURIForHeader(String name) {
+
+        String tenantQualifiedRelativePath =
+                String.format(TENANT_CONTEXT_PATH_COMPONENT, getTenantDomainFromContext()) + SERVER_API_PATH_COMPONENT;
+        String url = IdentityUtil.getEndpointURIPath(tenantQualifiedRelativePath + name, false, true);
+
+        URI loc = URI.create(url);
+        if (!loc.isAbsolute()) {
+            Message currentMessage = PhaseInterceptorChain.getCurrentMessage();
+            if (currentMessage != null) {
+                UriInfo ui = new UriInfoImpl(currentMessage.getExchange().getInMessage(), null);
+                try {
+                    return new URI(ui.getBaseUri().getScheme(), ui.getBaseUri().getAuthority(), url, null, null);
+                } catch (URISyntaxException e) {
+                    LOG.error("Server encountered an error while building the location URL with scheme: " +
+                            ui.getBaseUri().getScheme() + ", authority: " + ui.getBaseUri().getAuthority() +
+                            ", url: " + url, e);
+                }
+            }
+        }
+        return loc;
+    }
+
+    /**
+     * Retrieves loaded tenant domain from carbon context.
+     *
+     * @return tenant domain of the request being served.
+     */
+    private static String getTenantDomainFromContext() {
+
+        String tenantDomain = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
```

Comment:
if (IdentityUtil.threadLocalProperties.get().get(TENANT_NAME_FROM_CONTEXT) != null) {
            return (String) IdentityUtil.threadLocalProperties.get().get(TENANT_NAME_FROM_CONTEXT);
        }
        return MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;

Reply:
I think this is okay since it avoids multiple returns in the method.